extensions/amp-sidebar/1.0/sidebar.js

```diff
@@ -0,0 +1,216 @@
+/**
+ * Copyright 2020 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import * as Preact from '../../../src/preact';
+import {ContainWrapper} from '../../../src/preact/component';
+import {forwardRef} from '../../../src/preact/compat';
+import {setStyle} from '../../../src/style';
+import {
+  useCallback,
+  useEffect,
+  useImperativeHandle,
+  useRef,
+  useState,
+} from '../../../src/preact';
+import {useStyles} from './sidebar.jss';
+
+const ANIMATION_DURATION = 350;
+const ANIMATION_PRESETS = {
+  fadeIn: [{opacity: 0}, {opacity: 1}],
+  slideInLeft: [{transform: 'translateX(-100%)'}, {transform: 'translateX(0)'}],
```

Comment:
These names, most likely, should be in quotes. I.e. this is not an enum, but rather a simple string-keyed object map. Especially since you use string keys below to access this map. However, I'd also point out: do we really need a map here? Why not just have three plain constants?

Reply:
Good point, we can just use three constants for this.  Thanks!